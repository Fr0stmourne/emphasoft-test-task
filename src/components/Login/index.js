import React from 'react';
import { clientID, redirectCallbackUrl } from '../../utils/constants';
import PropTypes from 'prop-types';
import './index.scss';

export default function Login({
  title = 'Авторизация',
  buttonText = 'Авторизоваться Вконтакте',
}) {
  return (
    <div className="login">
      <h1 className="login__title">{title}</h1>
      <a
        className="login__button"
        href={`https://oauth.vk.com/authorize?client_id=${clientID}&display=page&redirect_uri=${redirectCallbackUrl}&scope=friends,offline&response_type=code&v=5.122`}
      >
        <span className="visually-hidden">{buttonText}</span>
      </a>
    </div>
  );
}

Login.propTypes = {
  title: PropTypes.string,
  buttonText: PropTypes.string,
};
