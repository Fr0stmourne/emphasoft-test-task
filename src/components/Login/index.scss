.login {
  &__button {
    display: block;
    width: 50px;
    height: 50px;
    background-image: url('./vk.svg');
    background-size: cover;
    margin: 0 auto;
  }
}