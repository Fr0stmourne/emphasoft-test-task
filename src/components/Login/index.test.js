import React from 'react';
import renderer from 'react-test-renderer';
import Login from '.';

describe('Friend component', () => {
  test('should render the component correctly when no props are provided', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render the component with custom props correctly', () => {
    const tree = renderer
      .create(<Login title="Auth" buttonText="Log in" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
