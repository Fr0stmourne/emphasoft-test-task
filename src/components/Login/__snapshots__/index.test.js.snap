// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friend component should render the component correctly when no props are provided 1`] = `
<div
  className="login"
>
  <h1
    className="login__title"
  >
    Авторизация
  </h1>
  <a
    className="login__button"
    href="https://oauth.vk.com/authorize?client_id=7560327&display=page&redirect_uri=http://localhost:3000/auth/vkontakte/callback&scope=friends,offline&response_type=code&v=5.122"
  >
    <span
      className="visually-hidden"
    >
      Авторизоваться Вконтакте
    </span>
  </a>
</div>
`;

exports[`Friend component should render the component with custom props correctly 1`] = `
<div
  className="login"
>
  <h1
    className="login__title"
  >
    Auth
  </h1>
  <a
    className="login__button"
    href="https://oauth.vk.com/authorize?client_id=7560327&display=page&redirect_uri=http://localhost:3000/auth/vkontakte/callback&scope=friends,offline&response_type=code&v=5.122"
  >
    <span
      className="visually-hidden"
    >
      Log in
    </span>
  </a>
</div>
`;
